### DISCUSSION TITLE
Help to understand a login error message

### DISCUSSION THREAD

**Discussion by sdespont** [symfony/symfony#59110] [Q&A] [answered]
My website is working fine, but I'm getting this error several times a day and I really don't understand why. 
Am I doing something wrong, like missing a validator, or is this a hacking attempt by bots?

`Uncaught PHP Exception Symfony\Component\HttpKernel\Exception\BadRequestHttpException: "The key "username" must be a string, "NULL" given." at .../vendor/symfony/security-http/Authenticator/FormLoginAuthenticator.php line 149`

Symfony 5.4.47

**Comment by alexisLefebvre** [comment]
Try to replicate the issue by sending a `POST` request without data to the log in endpoint, it may trigger this error.

I was able to do it on a local environment:

1. open the network tab
2. log in
3. right click on the `POST` request > choose something like "Change and send again" (in Firefox)
4. remove the body of the request
5. press “send”
6. you’ll see the exception

It may be bots that scan websites and do random requests.

**Comment by sdespont** [comment]
OK. Thanks. I also think it is bots. 
Any thoughts to ignore these alerts with monolog ?

**Reply by alexisLefebvre** [reply, marked as answer]
> OK. Thanks. I also think it is bots. Any thoughts to ignore these alerts with monolog ?

I didn’t test myself but it should work: https://symfony.com/doc/current/logging/monolog_exclude_http_codes.html

See also: https://github.com/symfony/symfony/issues/27961#issuecomment-916078720
